tests/connector/tests/hydrator_api_test.go

```diff
@@ -93,13 +96,33 @@ func TestHydrators(t *testing.T) {
 				}
 			}
 
-			//then
 			assert.Equal(t, true, hasAuth)
+
+			// check if the gql resolver is not sending used/expired auths
+			if testCase.clientType == "Application" {
+				appSystemAuths = fixtures.GetApplication(t, ctx, directorClient.DexGraphqlClient, cfg.Tenant, app.ID).Auths
+				assert.Len(t, appSystemAuths, 0)
+			}
 		})
 
 		t.Run("should resolve certificate for "+testCase.clientType, func(t *testing.T) {
 			//given
-			token, err := testCase.tokenGenerationFunc(t, testCase.clientId)
+			var err error
+			var token externalschema.Token
+
+			app, err := fixtures.RegisterApplicationFromInput(t, ctx, directorClient.DexGraphqlClient, cfg.Tenant, graphql.ApplicationRegisterInput{
```

Comment:
Why was it needed to move app creation for each test case and check inside the clientID which has to be passed to `tokenGenerationFunc`?